bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/types/BUnionType.java

```diff
@@ -55,55 +60,29 @@
     private static final String CLONEABLE = "Cloneable";
     private static final Pattern pCloneable = Pattern.compile(INT_CLONEABLE);
 
-    /**
-     * Create a {@code BUnionType} which represents the union type.
-     */
-    @Deprecated
-    public BUnionType() {
-        super(null, null, Object.class);
-        this.readonly = false;
-    }
-
-    /**
-     * Create a {@code BUnionType} which represents the union type.
-     *
-     * @param memberTypes of the union type
-     * @param typeFlags flags associated with the type
-     */
-    public BUnionType(List<Type> memberTypes, int typeFlags) {
-        this(memberTypes, typeFlags, false, false);
-    }
-
     public BUnionType(List<Type> memberTypes, int typeFlags, boolean readonly,  boolean isCyclic) {
         super(null, null, Object.class);
         this.typeFlags = typeFlags;
         this.readonly = readonly;
-        setMemberTypes(memberTypes);
+        this.flags = readonly ? SymbolFlags.READONLY : 0;
```

Comment:
Can we use this(memberTypes, memberTypes,typeFlags, isCyclic, readonly ? SymbolFlags.READONLY : 0)

Reply:
Fixed in 484ec9f, thanks!